### ISSUE TITLE
8.2.0 Installation issue

### ISSUE THREAD

**Issue by Tofandel** [salesagility/SuiteCRM-Core#178]
#### Issue
While installing SuiteCRM 8.2.0 an error occurs
![image](https://user-images.githubusercontent.com/6115458/200611225-574d7b80-f095-4fa1-8fa4-38acbad3b5a5.png)

The error message gives no clue of what's the issue, checked both logs and there is nothing in there

#### Your Environment

* SuiteCRM Version used: 8.2.0
* Environment name and version (e.g. MySQL, PHP 7): php 8.0.24 (All required php modules are installed), mariadb 10.7.7
* Operating System and version (e.g Ubuntu 16.04): Ubuntu 20.04

**Comment by Tofandel** [comment]
After selecting "Populate database with demo data" and removing the db port it finally worked..

That installation process is so painful

**Comment by pgorod** [comment]
Do you need help getting the ownerships and permissions right in Ubuntu? How did you set them?

Which user name is your web server running under? Is it `psacln`?

**Comment by Tofandel** [comment]
I chmoded them to 755 and 644 don't know why it's not the default

**Comment by pgorod** [comment]
Permissions always go together with ownerships - one only makes sense with the other. So please tell me also your `chown`

SuiteCRM  is an application that writes itself. When you change things in Studio, PHP code is produced and written to disk. So the permissions need to allow writing in ways that most applications just don't need.

Can you use the permissions commands recommended on the installation docs? You just need to figure out the correct user and group names (the ones that your web server is using to run under) and you should be ok.

**Comment by Tofandel** [comment]
The owner is correct, mainly because the files where all created during installation so by the webserver, but my point is, why does the install create public files in 770 and 660 and not in 755 and 644 like the installation doc suggest as a command to run (https://docs.suitecrm.com/admin/installation-guide/downloading-installing/) which is btw not a safe way to do this on the whole directory, as there is config files and env files in 660 which should stay in 660 otherwise they become visible to other users of the system but public directories should be 755 or apache will refuse to show them

The config.php file in legacy has those wrong 770 permissions harcoded in them after the installation, which I also changed after chmoding

I got there in the end after 3h of trial and (a lot of) errors

I'm just sharing my first install experience which was nightmarish

**Comment by pgorod** [comment]
That final zero is not a mistake, in fact it is better to use it, otherwise any user on your system will get access, which is not necessary and expands the area of possible insecurity, in an especially critical way on shared servers.

Permissions are more complex than this, there are still other factors we haven't considered here, which might justify the behaviors that were troublesome for you. This especially true for the ownerships and permissions assigned to new files.

Apache only refuses to serve files with a zero in the final digit of the permissions because that fileis owned by a user which is not the same as the user under which apache is running, and because it is owned by a group which does not include the user under which Apache is running.

I would recommend giving Apache access through a group, not the specific user, and to leave the final digit of permissions at 0.

**Comment by pgorod** [comment]
I can't comment on your specific install because you haven't provided all the information together - actual ownerships, actual permissions, actual web server user. It's better to check how it's actually running, not going by configuration files, since often the problem is getting the configs to become effective the way we expect them to.

> However it disallows web access to folders if they do not have read permission for all users (I think it doesn't care about files), this is how all plesk servers are configured, I don't know about others.

I find this surprising, there must be some other logic here. I don't see why plesk would make that kind of rule which reduces security.

> And it makes a lot of sense, if it's a web server it's publicly readable and so it's reasonable any user on the system can also list those files (but it's not reasonable they can write them) and if you disable any user access then it's reasonable to expect that web users should not be able to access those folders as well

Note that it's the web server **results** that are publicly readable, what the web server produces, not the files on the web server. It's a very different thing. For example, your `config.php` contains your database credentials in clear text. If it was publicly accessible it would be crazy insecure, and if it was somehow accessible to other users on a shared server it would also be terrible, especially on a shared server (though it's not _that_ simple to get there form another user account on a shared server, it's just an unnecessary weakening of security).

I do accept that much needs to be improved on SuiteCRM's side regarding permissions and installation experience. And the documentation. I'm just not convinced it's the changes and comments you're currently proposing.

But anyway, I'm glad you got it working, if you're happy with your set up, I am too :-)

Peace.

**Comment by Tofandel** [comment]
> Note that it's the web server results that are publicly readable, what the web server produces, not the files on the web server. It's a very different thing. For example, your config.php contains your database credentials in clear text. If it was publicly accessible it would be crazy insecure, and if it was somehow accessible to other users on a shared server it would also be terrible, especially on a shared server (though it's not that simple to get there form another user account on a shared server, it's just an unnecessary weakening of security).

Yes I'm aware, and we can agree on that, that's why I always set those config files in `600` (after install, the .env.local was in 664, that's a huge security issue, it seems to ignore any setting and will just always create it like this)

I would prefer `750` and `640` in general for all files and 600 for sensitive config files

I had a look at how plesk is handling all this, and it does seem that apache is supposed to be running as `www-data:www-data` and su exec changes it to `tukan-dev:psacln`, I don't know how this all works under the hood but it does seem that like this apache can't access the psacln folders in `750` unless `www-data` is assigned to the `psacln` group, so that's what I did, not sure why plesk doesn't do that by default

Anyways, from a security standpoint the installation procedure needs to be revised that's for sure, permissions were all over the place (I only chmoded the `legacy/cache` folder)

I can redo one install from scratch and I'll report the invalid sensitive permissions I can find

**Comment by pgorod** [comment]
Now _that_ seems to help start making sense of what happened. Thanks for taking the trouble to come back here and explain.

If you do get around to redoing an install from scratch, for us it would be most useful to start with the commands recommended in the docs: 
```
find . -type d -not -perm 2755 -exec chmod 2755 {} \;
find . -type f -not -perm 0644 -exec chmod 0644 {} \;
find . ! -user www-data -exec chown www-data:www-data {} \;
chmod +x bin/console
```

Although it's ok to change the final digits to 0 (as long as Apache successfully gets access as "user" or "group", not "world") and to change www-data to pscln (as long as Apache belongs to that group). That `2` in the directory permissions is also important.

Thanks

**Comment by Tofandel** [comment]
@pgorod Where did you find those commands?

This is what the doc says for me, hence my previous comments
```
  sudo chown -R www-data:www-data .
  sudo chmod -R 755 .
  sudo chmod -R 775 cache custom modules themes data upload
  sudo chmod 775 config_override.php 2>/dev/null
```

**Comment by Tofandel** [comment]
Seems like it, there is no version written anywhere on the page
https://docs.suitecrm.com/admin/installation-guide/downloading-installing/#_copying_suitecrm_files_to_web_server

**Comment by pgorod** [comment]
Fixed! 

https://github.com/salesagility/SuiteDocs/pull/585
(waiting to be merged)

There was a notice in the v8 page explaining that there was a separate page for v7. But not the other way around 🤷
